Rollbar/DTOs/Body.cs

```diff
@@ -0,0 +1,116 @@
+﻿namespace Rollbar.DTOs
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Newtonsoft.Json;
+    using Rollbar.Diagnostics;
+
+    public class Body
+        : DtoBase
+    {
+        public Body(IEnumerable<System.Exception> exceptions)
+        {
+            if (exceptions == null)
+            {
+                throw new ArgumentNullException(nameof(exceptions));
+            }
+
+            var allExceptions = exceptions as System.Exception[] ?? exceptions.ToArray();
+            if (!allExceptions.Any())
+            {
+                throw new ArgumentException("Trace Chains must have at least one Trace", nameof(exceptions));
+            }
+
+            TraceChain = allExceptions.Select(e => new Trace(e)).ToArray();
+
+            Validate();
+        }
+
+        public Body(AggregateException exception) 
+            : this(exception.InnerExceptions)
+        {
+        }
+
+        public Body(System.Exception exception)
+        {
+            if (exception == null)
+            {
+                throw new ArgumentNullException(nameof(exception));
+            }
+
+            if (exception.InnerException != null)
+            {
+                var exceptionList = new List<System.Exception>();
+                var outerException = exception;
+                while (outerException != null)
+                {
+                    exceptionList.Add(outerException);
+                    outerException = outerException.InnerException;
+                }
+                TraceChain = exceptionList.Select(e => new Trace(e)).ToArray();
+            }
+            else
+            {
+                Trace = new Trace(exception);
+            }
+
+            Validate();
+        }
+
+        public Body(Message message)
+        {
+            Message = message;
+            Validate();
+        }
+
+        public Body(string crashReport)
+        {
+            this.CrashReport = new CrashReport(crashReport);
```

Comment:
The API type of crash_report is only used for iOS at the moment as far as I know. So this might end up being unused here. It isn't really hurting being here, but we should probably remove it if it really is completely unused.

Reply:
As far as I know, Microsoft is moving more towards wider cross-platform development: .NET Core and Visual Studio for Mac with Xamarin support on macOS and iOS are strong signs of it. So, they might support crash reports as well (or maybe even now, since I am not an expert on Xamarin yet).
Maybe we can keep it since we already have it?